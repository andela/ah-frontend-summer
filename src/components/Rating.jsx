import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import StarRatingComponent from 'react-star-rating-component';

import '../styles/Articles.scss';

const Rating = props => {
    const {
        rateScore,
        onStarClick,
        averageRating
    } = props;

    const isLoggedIn = localStorage.getItem('token');

    return (
        <div>
            {isLoggedIn ? (
                <Container textAlign="right">
                    <StarRatingComponent
                        name="rating"
                        value={rateScore}
                        starCount={5}
                        starColor="#008080"
                        emptyStarColor="grey"
                        onStarClick={onStarClick}
                        className="rating-stars"
                    />
                </Container>
            ) : ""}

            <Container textAlign="right">
                <span>
                    Average Rates:
                    {averageRating}
                </span>
            </Container>

        </div>
    );
};
Rating.defaultProps = {
    rateScore: 0,
    averageRating: 0,
    onStarClick: () => {}
};

Rating.propTypes = {
    rateScore: PropTypes.number,
    onStarClick: PropTypes.func,
    averageRating: PropTypes.number
};

export default Rating;
