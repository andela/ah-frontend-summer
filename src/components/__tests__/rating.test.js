import React from 'react';
import { mount } from "enzyme";

import Rating from '../Rating';

describe('Rating', () => {
    const onStarClick = jest.fn();
    const ratingProps = {
        rateScore: 4,
        onStarClick: onStarClick,
        averageRating: 4
    };
    const rate = mount(<Rating {...ratingProps} />);
    it("should render rate component correctly", () => {
        expect(rate).toMatchSnapshot();
    });
});
