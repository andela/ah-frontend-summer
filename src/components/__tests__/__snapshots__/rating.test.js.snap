// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render rate component correctly 1`] = `
<Rating
  averageRating={4}
  onStarClick={[MockFunction]}
  rateScore={4}
>
  <div>
    <Container
      textAlign="right"
    >
      <div
        className="ui right aligned container"
      >
        <span>
          Average Rates:
          4
        </span>
      </div>
    </Container>
  </div>
</Rating>
`;
